ocs_ci/deployment/ibmcloud.py

```
@@ -774,3 +790,705 @@ def prepare_custom_vpc_and_network(self):
+            )
+            exec_cmd(cmd)
+            logger.info(
+                f"Successfully added security group '{security_group_name}' to instance "
+                f"{instance_name} network interface {nic_name}"
+            )
+            return True
+
+        except Exception as e:
+            logger.error(f"Failed to add security group to VSI: {e}")
+            return False
+
+    def _wait_for_resource_deletion(
+        self, resource_type, resource_id, max_attempts=30, wait_time=5
+    ):
+        """
+        Wait for resource deletion with retry.
+
+        Args:
+            resource_type (str): The type of resource to check for deletion
+                (e.g., "instance", "load-balancer", "volume", etc.).
+            resource_id (str): The unique identifier of the resource to check.
+            max_attempts (int, optional): Maximum number of attempts to check for deletion. Default is 30.
+            wait_time (int, optional): Time in seconds to wait between checks. Default is 5.
+
+        Returns:
+            bool: True if the resource is deleted within the specified attempts, False otherwise.
+        """
+        for attempt in range(1, max_attempts + 1):
+            try:
+                if resource_type == "instance":
+                    cmd = f"ibmcloud is instance {resource_id} --output json"
+                elif resource_type == "load-balancer":
+                    cmd = f"ibmcloud is load-balancer {resource_id} --output json"
+                elif resource_type == "volume":
+                    cmd = f"ibmcloud is volume {resource_id} --output json"
+                elif resource_type == "security-group":
+                    cmd = f"ibmcloud is security-group {resource_id} --output json"
+                elif resource_type == "floating-ip":
+                    cmd = f"ibmcloud is floating-ip {resource_id} --output json"
+                else:
+                    return True
```

We should rather raise an exception of unsupported resource type here